client/functions.lua

```diff
@@ -177,29 +178,36 @@ function QBCore.Functions.GetPlayersFromCoords(coords, distance)
     return closePlayers
 end
 
-function QBCore.Functions.GetClosestVehicle()
-    local ped = PlayerPedId()
-    local coords = GetEntityCoords(ped)
-    local vehicles = GetGamePool('CVehicle')
-    local closestDistance = -1
-    local closestVehicle = -1
-    for i = 1, #vehicles, 1 do
-        local vehicleCoords = GetEntityCoords(vehicles[i])
-        local distance = #(vehicleCoords - coords)
-        if closestDistance == -1 or closestDistance > distance then
-            closestVehicle = vehicles[i]
-            closestDistance = distance
-        end
-    end
-    return closestVehicle, closestDistance
+function QBCore.Functions.GetClosestVehicle(coords)
+	local ped = PlayerPedId()
+	local vehicles        = GetGamePool('CVehicle')
+	local closestDistance = -1
+	local closestVehicle  = -1
+	local coords          = coords
+	if coords == nil then
+		coords = GetEntityCoords(ped)
+	end
+	for i=1, #vehicles, 1 do
+		local vehicleCoords = GetEntityCoords(vehicles[i])
+		local distance = #(vehicleCoords - coords)
+
+		if closestDistance == -1 or closestDistance > distance then
+			closestVehicle  = vehicles[i]
+			closestDistance = distance
+		end
+	end
+	return closestVehicle,closestDistance
 end
 
-function QBCore.Functions.GetClosestObject()
+function QBCore.Functions.GetClosestObject(coords)
     local ped = PlayerPedId()
-    local coords = GetEntityCoords(ped)
     local objects = GetGamePool('CObject')
     local closestDistance = -1
     local closestObject = -1
+    local coords          = coords
```

Comment:
Same here

Reply:
Actually, I want to eliminate the "or" possibility by making a check here. If it works stably, of course I add it as "or".

Maybe it's better to do it this way.
`if not coords then coords = GetEntityCoords(ped) end`